src/core/output.rs

```diff
@@ -1,40 +1,24 @@
-use std::{path::Path, process::Command};
+use flate2::{write::GzEncoder, Compression};
+use std::{fs::File, path::Path};
+use tar::Builder;
 
-use super::{CliError, LalResult};
+use super::{LalResult};
 
 /// Helper for stash and build
 pub fn tar(component_dir: &Path, tarball: &Path) -> LalResult<()> {
     info!("Taring OUTPUT");
-    let mut args: Vec<String> = vec![
-        "czf".into(),
-        tarball.to_str().unwrap().into(), // path created internally - always valid unicode
-        "--transform=s,^OUTPUT/,,".into(), // remove leading OUTPUT
-    ];
 
-    // Avoid depending on wildcards (which would also hide hidden files)
-    // All links, hidden files, and regular files should go into the tarball.
-    let findargs = vec!["OUTPUT/", "-type", "f", "-o", "-type", "l"];
-    debug!("find {}", findargs.join(" "));
-    let find_output = Command::new("find")
-        .args(&findargs)
-        .current_dir(&component_dir)
-        .output()?;
-    let find_str = String::from_utf8_lossy(&find_output.stdout);
+    let tarball = File::create(tarball)?;
+    let compressor = GzEncoder::new(tarball, Compression::default());
+    let mut archive = Builder::new(compressor);
 
-    // append each file as an arg to the main tar process
-    for f in find_str.trim().split('\n') {
-        args.push(f.into())
-    }
+    // Don't dereference symlinks, archive them as-is.
+    // Dereferencing symlinks makes the archive larger, as we will then store two copies  of the
+    // same file. Additionally, if OUTPUT contains dangling symlinks, adding them to the archive
+    // will fail with a NotFound error.
+    archive.follow_symlinks(false);
```

Comment:
Ok. This sounds good. (Had to re-read this a few times as it made it sound like we were doubling our tarball sizes in some cases.)

So we only allow: symlinks that point to something relative to (and within) OUTPUT.
And everything else is the same when you inspect a tarball with `tar`?

Reply:
Hmm, we should definitely add a check (and enforce with tests) to ensure that symlinks can only be relative to OUTPUT.

I've definitely managed to accidentally create absolute symlinks in local testing.
My current workaround was to write a `rewrite_symlinks.sh` script, but perhaps we should fail the build (or `build --release` at the very least) step.